src/api/use-cart.ts

```diff
@@ -0,0 +1,115 @@
+import type { CartItem, CartTotals } from './types';
+import { useEcomAPI } from './ecom-api-context-provider';
+import { cart as cartType } from '@wix/ecom';
+import { getCartItemSubtotal } from './cart-helpers';
+import {
+    useAddToCart,
+    useCartTotals,
+    useRemoveItemFromCart,
+    useUpdateCartItemQuantity,
+} from './api-hooks';
+import { media } from '@wix/sdk';
+import { useState } from 'react';
+
+export interface UseCartResult {
+    cartTotals?: CartTotals;
+    cartItems: CartItem[];
+    isAddingToCart: boolean;
+    isCartTotalUpdating: boolean;
+    addToCart: (args1: any, args2: any) => void; //  any temporarily
+    handleCheckout: () => void;
+}
+
+export const useCart = (): UseCartResult => {
+    const ecomAPI = useEcomAPI();
+    const { data: cartData } = useCartTotals();
+    const [updatingCartItemIds, setUpdatingCartItemIds] = useState<string[]>([]);
```

Comment:
These are cart items with loader which are waiting while Cart Summary updating finishes. Those items and Cart summary will simultaneously remove their loaders.